pkg/internal/computeresources/transformer.go

```diff
@@ -100,24 +100,38 @@ func transformPodBasedOnLimitRange(p *corev1.Pod, limitRange *corev1.LimitRange)
 			p.Spec.Containers[i].Resources.Requests = defaultRequests
 		} else {
 			for _, name := range resourceNames {
-				setRequestsOrLimits(name, p.Spec.Containers[i].Resources.Requests, defaultRequests)
+				setRequests(name, p.Spec.Containers[i].Resources.Requests, defaultRequests, defaultContainerLimits)
 			}
 		}
 		if p.Spec.Containers[i].Resources.Limits == nil {
 			p.Spec.Containers[i].Resources.Limits = defaultContainerLimits
 		} else {
 			for _, name := range resourceNames {
-				setRequestsOrLimits(name, p.Spec.Containers[i].Resources.Limits, defaultContainerLimits)
+				setLimits(name, p.Spec.Containers[i].Resources.Limits, defaultContainerLimits)
 			}
 		}
 	}
 	return p
 }
 
-func setRequestsOrLimits(name corev1.ResourceName, dst, src corev1.ResourceList) {
-	if compare.IsZero(dst[name]) && !compare.IsZero(src[name]) {
-		dst[name] = src[name]
+// setRequests sets the requests to the max of (currentRequests, defaultRequests)
+// and the min of (currentRequests, defaultLimits)
```

Comment:
I think what actually needs to happen is that the requests are:
- greater than the min request
- smaller than the max limit
- set to the requests if they exist
- smaller than whatever limit we end up with

Using default limits here is not right if we have both requests and limits > default limits-- in that case we'd just want to use the specified requests and limits.

Also, this docstring reflects details of how the function is called (i.e. the "limits" are the default limits of the limitRange); could you reword it to focus on what the function does?

Reply:
I see. thanks for details about the logic. 

so requests will be ensured to have a value larger than Min of LimitRange while leaving limits set by the lower end of limits (of container) and LimitRange.